src/acquisition/flusurv/flusurv_update.py

```diff
@@ -113,38 +112,185 @@ def update(issue, location_name, test_mode=False):
-      `rate_age_3` = coalesce(%s, `rate_age_3`),
-      `rate_age_4` = coalesce(%s, `rate_age_4`),
-      `rate_overall` = coalesce(%s, `rate_overall`),
-      `rate_age_5` = coalesce(%s, `rate_age_5`),
-      `rate_age_6` = coalesce(%s, `rate_age_6`),
-      `rate_age_7` = coalesce(%s, `rate_age_7`)
+      `release_date` = least(`release_date`, %(release_date)s),
+      `rate_overall` = coalesce(%(rate_overall)s, `rate_overall`),
+
+      `rate_age_0` = coalesce(%(rate_age_0)s, `rate_age_0`),
+      `rate_age_1` = coalesce(%(rate_age_1)s, `rate_age_1`),
+      `rate_age_2` = coalesce(%(rate_age_2)s, `rate_age_2`),
+      `rate_age_3` = coalesce(%(rate_age_3)s, `rate_age_3`),
+      `rate_age_4` = coalesce(%(rate_age_4)s, `rate_age_4`),
+      `rate_age_5` = coalesce(%(rate_age_5)s, `rate_age_5`),
+      `rate_age_6` = coalesce(%(rate_age_6)s, `rate_age_6`),
+      `rate_age_7` = coalesce(%(rate_age_7)s, `rate_age_7`),
+
+      `rate_age_18t29` = coalesce(%(rate_age_18t29)s, `rate_age_18t29`),
+      `rate_age_30t39` = coalesce(%(rate_age_30t39)s, `rate_age_30t39`),
+      `rate_age_40t49` = coalesce(%(rate_age_40t49)s, `rate_age_40t49`),
+      `rate_age_5t11` = coalesce(%(rate_age_5t11)s, `rate_age_5t11`),
+      `rate_age_12t17` = coalesce(%(rate_age_12t17)s, `rate_age_12t17`),
+      `rate_age_lt18` = coalesce(%(rate_age_<18)s, `rate_age_lt18`),
+      `rate_age_gte18` = coalesce(%(rate_age_>=18)s, `rate_age_gte18`),
+
+      `rate_race_white` = coalesce(%(rate_race_white)s, `rate_race_white`),
+      `rate_race_black` = coalesce(%(rate_race_black)s, `rate_race_black`),
+      `rate_race_hisp` = coalesce(%(rate_race_hispaniclatino)s, `rate_race_hisp`),
+      `rate_race_asian` = coalesce(%(rate_race_asianpacificislander)s, `rate_race_asian`),
+      `rate_race_natamer` = coalesce(%(rate_race_americanindianalaskanative)s, `rate_race_natamer`),
+
+      `rate_sex_male` = coalesce(%(rate_sex_male)s, `rate_sex_male`),
+      `rate_sex_female` = coalesce(%(rate_sex_female)s, `rate_sex_female`)
     """
 
     # insert/update each row of data (one per epiweek)
     for epiweek in epiweeks:
         lag = delta_epiweeks(epiweek, issue)
-        if lag > 52:
+        if lag > max_age_to_consider_weeks:
             # Ignore values older than one year, as (1) they are assumed not to
             # change, and (2) it would adversely affect database performance if all
             # values (including duplicates) were stored on each run.
             continue
-        args_meta = [release_date, issue, epiweek, location, lag]
-        args_insert = data[epiweek]
-        args_update = [release_date] + data[epiweek]
-        cur.execute(sql, tuple(args_meta + args_insert + args_update))
+
+        # As of Sept 2023, for new data we expect to see these 23 groups, as
+        #  described in the top-level "master_lookup" element of the new GRASP API
+        #  (https://gis.cdc.gov/GRASP/Flu3/PostPhase03DataTool) response object:
```

Comment:
a few things about this comment block:
 * its better to make it a footnote with the bulk of the text somewhere else in this file instead of interrupting the method (and loop!) with a huge comment.  maybe even make `n_expected_groups` into a top-level constant?
 * it might be easier to read as an unassigned multi-line string instead of using the octothorpe prefix on each line
 * its probably worth it to mention your sample json files in `./reference/`